tests/integration/test_merge_tree_s3/test.py

```diff
@@ -562,6 +558,13 @@ def test_s3_disk_apply_new_settings(cluster, node_name):
     node = cluster.instances[node_name]
     create_table(node, "s3_test")
 
+    config_path = os.path.join(
```

Comment:
I think there is no point moving it from global variable to local. It is not unique per test and quite high chance it will be needed again.

Reply:
You need to move `cluster = ClickHouseCluster(__file__)` to global scope then.
The logic is that `cluster.instances_dir_name` is a part of a cluster. 
`get_instances_dir()` was used in couple of tests and it was incorrect at least for clusters what have `name`.